lib/Driver/Collection/CollectionAbstractField.php

```diff
@@ -45,14 +46,14 @@ public function buildCriteria(Criteria $c)
         $comparison = $this->getComparison();
         $eb = Criteria::expr();
 
-        if ($comparison == 'between') {
+        if ('between' === $comparison) {
             if (false === is_array($data)) {
                 throw new CollectionDriverException(
                     'Fields with \'between\' comparison require to bind an array.'
                 );
             }
 
-            $from = count($data) ? array_shift($data) : null;
+            $from = array_shift($data);
             $to = count($data) ? array_shift($data) : null;
 
             if (!$from && ($from !== 0)) {
```

Comment:
When using `!`, isn't 0 included in it as well? I think the second part of the condition may be redundant.